sematic/config/settings.py

```diff
@@ -3,193 +3,193 @@
-
-        if value is not None:
-            return str(value)
-
-        if len(args) >= 1:
+
+def get_active_settings() -> ProfileSettings:
+    global _SETTINGS
+
+    if _SETTINGS is None:
+        settings = get_settings()
+
+        profile_settings = settings.profiles[DEFAULT_PROFILE]
+
+        for plugin_path, plugin_settings in profile_settings.settings.items():
+            plugin_settings_vars = _get_plugin_settings_vars(plugin_path)
+            _apply_env_var_overrides(plugin_settings, plugin_settings_vars)
+
+    return get_settings().profiles[DEFAULT_PROFILE]
+
+
+def get_active_plugins(
+    scope: PluginScope, default: List[Type[AbstractPlugin]]
+) -> List[Type[AbstractPlugin]]:
+    scopes = get_active_settings().scopes
+
+    if scope not in scopes:
+        return default
+
+    imported_plugins = [import_plugin(plugin_path) for plugin_path in scopes[scope]]
+
+    return imported_plugins
+
+
+def get_plugin_settings(
+    plugin: Type[AbstractPlugin],
+) -> Dict[AbstractPluginSettingsVar, str]:
+    settings = get_active_settings().settings
+
+    plugin_path = plugin.get_path()
+
+    if plugin_path not in settings:
+        raise MissingSettingsError(plugin)
+
+    return settings[plugin_path]
+
+
+def get_plugin_setting(
+    plugin: Type[AbstractPlugin], var: AbstractPluginSettingsVar, *args
+) -> str:
+    try:
+        plugin_settings = get_plugin_settings(plugin)
+    except MissingSettingsError:
+        plugin_settings = {}
+
+    if var not in plugin_settings:
+        if len(args) > 0:
             return args[0]
 
-        raise MissingSettingsError(var, self.cli_command)
+        raise MissingSettingsError(plugin, var)
 
-    def set_setting(self, var: AbstractSettingsVar, value: str) -> None:
-        """
-        Set value for setting.
+    return plugin_settings[var]
 
-        Parameters
-        ----------
-        var: AbstractSettingsVar
-            setting to set
-        value: str
-            value to set
-        """
-        if var not in self.vars:
-            raise ValueError(f"Unknown setting {var.value}")
 
-        settings = self.get_settings()
+def import_plugins():
+    scopes = get_active_settings().scopes
 
-        settings[DEFAULT_PROFILE][var] = value
+    for selected_plugins in scopes.values():
+        for plugin_path in selected_plugins:
+            import_plugin(plugin_path)
 
-        self.save_settings(settings)
 
-        # Void cache to force refresh
-        self._active_settings = None
+def set_plugin_setting(
+    plugin: Type[AbstractPlugin], var: AbstractPluginSettingsVar, value: str
+):
+    try:
+        plugin_settings = get_plugin_settings(plugin)
+    except MissingSettingsError:
+        plugin_settings = {}
+        get_active_settings().settings[plugin.get_path()] = plugin_settings
 
-    def delete_setting(self, var: AbstractSettingsVar) -> None:
-        """
-        Delete setting value.
+    plugin_settings[var] = value
```

Comment:
When you write a settings var, can you also ensure that there's a `__version__` field for the plugin in the config file that records what version of the plugin the settings correspond to?

Reply:
How will plugin developers specify migrations for their plugins' specific configs?

Reply:
Done.

Reply:
> How will plugin developers specify migrations for their plugins' specific configs?

there is no mechanism for that yet, we can add it later. But it will be easier to write if it's guaranteed there are already versions in people's config files.